### PR TITLE
adding redis common commands

### LINKED ISSUE #15: implement most common commands as temporary placeholders
As discussed in #13 , we;ll be implementing a couple of commonly used commands by hand, temporarily. Read the issue comments for better explanation of why.

Based on a couple of redis tutorials (and chatgpt of course), it seems to me that the queue/cache functions we should focus for now are"

- LPUSH
- RPUSH
- LPOP
- RPOP
- BRPOP
- BLPOP
- LRANGE
- LINDEX
- LSET
- LREM
- LLEN
- LINSERT
- LTRIM
- RPOPLPUSH
- BRPOPLPUSH
- LMOVE

### PR DESCRIPTION
**Pull request by Felipe-gsilva**
fixes #15

adds:
- most common redis command into clj-rq
- tests for these commands

## Summary by CodeRabbit

- **New Features**
	- Added comprehensive project documentation with a new README file, enhancing user understanding and engagement.
	- Introduced multiple new functions for queue management, including `bpop!`, `lindex`, `lset`, `lrem`, and more, expanding operational capabilities.

- **Bug Fixes**
	- Improved documentation for existing functions to clarify usage and enhance user guidance.

- **Tests**
	- Expanded test coverage for new queue functionalities, ensuring robustness and correctness of operations.
	- Enhanced organization of test cases for improved readability and clarity.

### CODE REVIEW

**Comment by avelino** [comment]
@coderabbitai review

**Comment by avelino** [comment]
@J0sueTM I believe a review is missing here, can you do it?
I know it will be replaced with this other PR
https://github.com/moclojer/clj-rq/pull/18

**Comment by J0sueTM** [comment]
Of course. It will not be replaced by the linked PR, but we'll rather use the tests written by @Felipe-gsilva

File: docs/README.md
Comment (J0sueTM) [reply docs/README.md:1]: @coderabbitai please write a compelling README content. It should expand the existing root README, but definitely not be the same as the the former.

Comment (J0sueTM) [reply docs/README.md:1]: @coderabbitai Thanks.

@avelino Take a look at this README. I think it could be the root README. What do you think?

File: src/com/moclojer/rq/queue.clj
Comment (J0sueTM) [inline src/com/moclojer/rq/queue.clj:190]: @Felipe-gsilva This does the same operation thrice. Parse a clojure seq to string, them read that string back to a clojure seq, and then parse it back to a string??

It's better to just use clj's builtin `pr-str` helper. Anyways, I won't request any changes (if I did, this would have to be improved all over the changes) because the implementation won't be used as soon as we merge the dynamic fns queue.

File: docs/README.md
Comment (Felipe-gsilva) [reply docs/README.md:1]: @J0sueTM the doc for this is set on #14, if you want to take a look

File: src/com/moclojer/rq/queue.clj
Comment (J0sueTM) [reply src/com/moclojer/rq/queue.clj:190]: Great!

File: docs/README.md
Comment (J0sueTM) [reply docs/README.md:1]: @Felipe-gsilva my bad lol. Thanks for the link.

@coderabbitai the main branch already has the doc I asked you to write, so there's no need for it afterall.

Resolving the conversation.

**Comment by J0sueTM** [comment]
@Felipe-gsilva i fixed the merge conflict.

Please pull and manually merge this branch to main when possible.

File: src/com/moclojer/rq/queue.clj
Comment (avelino) [reply src/com/moclojer/rq/queue.clj:7]: @Felipe-gsilva this working?

File: docs/README.md
Comment (J0sueTM) [reply docs/README.md:31]: @coderabbitai not necessary

File: src/com/moclojer/rq/queue.clj
Comment (Felipe-gsilva) [reply src/com/moclojer/rq/queue.clj:7]: @avelino Yeah, working, but now it is a little bit better. Forever learning, I guess LOL.

**Review by J0sueTM — APPROVED** [review]
LGTM!

**Pull request merged**
